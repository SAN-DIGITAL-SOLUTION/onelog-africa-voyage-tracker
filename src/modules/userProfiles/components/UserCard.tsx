import React from 'react';

export type UserCardProps = {
  id: string;
  name: string;
  role: string;
  onEdit?: () => void;
};

export const UserCard: React.FC<UserCardProps> = ({ id, name, role, onEdit }) => (
  <div style={{
    border: '1px solid #eee',
    borderRadius: 8,
    padding: 16,
    marginBottom: 12,
    background: '#fff',
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'center',
  }}>
    <div>
      <div style={{ fontWeight: 700 }}>{name}</div>
      <div style={{ color: 'var(--color-secondary)' }}>{role}</div>
    </div>
    {onEdit && (
      <button onClick={onEdit} style={{ background: 'var(--color-accent-cta)', color: '#fff', border: 'none', borderRadius: 4, padding: '6px 14px', cursor: 'pointer' }}>
        Modifier
      </button>
    )}
  </div>
);
