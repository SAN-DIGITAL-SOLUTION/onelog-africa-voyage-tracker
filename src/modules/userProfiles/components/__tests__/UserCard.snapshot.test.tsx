import { render } from '@testing-library/react';
import { UserCard } from '../UserCard';

describe('UserCard snapshot', () => {
  it('matches the snapshot', () => {
    const { container } = render(
      <UserCard user={{ id: '1', name: 'Jane Doe', role: 'admin' }} />
    );
    expect(container).toMatchSnapshot();
  });
});
