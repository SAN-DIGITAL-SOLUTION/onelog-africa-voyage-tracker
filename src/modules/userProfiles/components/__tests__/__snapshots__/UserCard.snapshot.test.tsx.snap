// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UserCard snapshot > matches the snapshot 1`] = `
<div>
  <div
    style="border: 1px solid rgb(238, 238, 238); border-radius: 8px; padding: 16px; margin-bottom: 12px; background: rgb(255, 255, 255); display: flex; justify-content: space-between; align-items: center;"
  >
    <div>
      <div
        style="font-weight: 700;"
      />
      <div
        style="color: var(--color-secondary);"
      />
    </div>
  </div>
</div>
`;
